src/rust/src/x509/sct.rs

```diff
@@ -139,13 +254,29 @@ pub(crate) fn parse_scts(
         let log_id = sct_data.read_exact(32)?.try_into().unwrap();
         let timestamp = u64::from_be_bytes(sct_data.read_exact(8)?.try_into().unwrap());
         let _extensions = sct_data.read_length_prefixed()?;
-        let _sig_alg = sct_data.read_exact(2)?;
-        let _signature = sct_data.read_length_prefixed()?;
+        let hash_algorithm = sct_data.read_byte()?.try_into()?;
+        if hash_algorithm == HashAlgorithm::None {
```

Comment:
That was what I originally proposed to @reaperhulk, so that we wouldn't need the additional test vectors 🙂 

IIRC the justification was that the CT RFC doesn't actually forbid these, it's just not clear what their semantics are. So we should support the enums as described in the spec and restrict during construction. But I'm happy to change it.

Reply:
Works for me. I can also go ahead and remove the new testvectors, if you'd like.

Reply:
Let's keep the vectors -- the same tests should pass for them since they're just "unknown algorithms" now right?

Reply:
Actually, ignore that -- we can still use the testvectors for their specific exception messages. But I'll remove the variants.

Reply:
[45c8fcd](https://github.com/pyca/cryptography/pull/7207/commits/45c8fcd5fa5d6ed872cd5ddd25aed0d4ac34ab20)